pkg/operator/ceph/version/version.go

```diff
@@ -106,7 +112,17 @@ func ExtractCephVersion(src string) (*CephVersion, error) {
 		return nil, fmt.Errorf("failed to parse version extra part: %s", m[2])
 	}
 
-	return &CephVersion{major, minor, extra}, nil
+	// See if we are running on a build release
+	mm := buildVersionPattern.FindStringSubmatch(src)
+	// We don't need to handle any error here, so let's jump in only when "mm" has content
+	if mm != nil {
+		build, err = strconv.Atoi(mm[4])
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse version build number part: %q", mm[3])
```

Comment:
should this be `mm[4]` in the error message? Or should `mm[3]` be parsed in the previous line?

Reply:
is this critical to fail? Or can we just log the error and keep the build field as 0?

Reply:
I re-used the current logic which seems to prevent an out of range, if we can't parse `mm[4]`, this likely because the index is out of range, so we print `mm[3]` to show what was the last value.

As for the error, we could live the default value of build I suppose but since we got a match, we are expected to its value, so if `strconv.Atoi`(which is unlikely) I'd like to preserve its error.

Reply:
If the regex succeeds, looks like `mm[4]` should be in range. If it was out of range, the operator would crash. How about a unit test that will check parsing the version string? The integration tests won't hit this code path upstream.

Reply:
No, the op wouldn't crash because a non-match means `mm == nil` and we don't handle that case on purpose but yes it looks like a bug, you're right.

Reply:
I see now the unit test already covers this. The crash i was referring to was a different case, but in any case we're covered by the regex match